### ISSUE TITLE
Jedi doesn't complete uninit annotated field 

### ISSUE THREAD

**Issue by qmmp123** [davidhalter/jedi#1847]
```python
class Base:
    foobar: int

class BaseMixin():
    field = Base

BaseMixin.field.# <- here i want to get foobar in list of completions
```

I think it is bug. Can you help me to solve it? I want to make PR to fix it. Where should I start? I've installed jedi from source code but I don't understand when it creates completions

**Comment by qmmp123** [comment]
I've found code
```python

**Comment by davidhalter** [comment]
This is very much intentional. I have always assumed that you should write `foobar: ClassVar[int]`. After reading a couple PEPs and mypy discussions I still think that way. It's kind of weird to suggest names that are defined and used as instance variables.

**Comment by qmmp123** [comment]
Oh, I'm so sorry. I did not see PEP526

**Comment by qmmp123** [comment]
I want to reopen this Issue but with another one example:
```Python
from typing import ClassVar


class Foo:
    field = None

    def __init__(self, field=None):
        self.field = field


class Bar(Foo):
    field: ClassVar[int] = 0

    def some_method(self):
        self.field.# <- here i got completions for None but I want to get completions for int
```

Sorry if the question is off topic but I did not find anything about this

**Comment by davidhalter** [comment]
Hmm good question. I think that it's more correct to access Bar.field if it's a class var, but I tend to agree that it should probably be among the instance completions.

Also mypy is perfectly fine with accessing `self.field`, so I guess we should complete it there.

**Comment by qmmp123** [comment]
Can I help you with it?

**Comment by qmmp123** [comment]
@davidhalter I've fixed my issue but there is one problem with `test/test_utils.py::TestSetupReadline::test_import`. I don't understand what this test checks. Can you see my PR?

**Comment by qmmp123** [comment]
@davidhalter Can you give me your feedback about my PR please?

**Comment by davidhalter** [comment]
I have merged your PR and then after playing with it decided that it was probably not the right approach. I removed the initial ClassVar filtering and think that's the right way.

I then also removed one of your tests, because I feel like the test is kind of confusing and shouldn't necessarily return `int()`. However I guess most of your cases should now be supported.

Thanks for contributing and sorry for the huge delay.
